pkg/component/autoscaling/gardenercustommetrics/kubeobjects/service.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2024 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeobjects
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+)
+
+func makeService(namespace string) *corev1.Service {
+	//This service intentionally does not contain a pod selector. As a result, KCM does not perform any endpoint management.
+	//Endpoint management is instead done by the gardener-custom-metrics leader instance, which ensures a single endpoint,
+	//directing all traffic to the leader.
```

Comment:
https://github.com/gardener/gardener/commit/2ac1580ed035a909c12299d77e2892695c3f58d9#diff-9da288ed98d152ea7f72c0d36b688c8db5ad36356e94f310b32f58ff766414c3R28-R30

Reply:
Done - the function was renamed.

Reply:
My cosmetic comment was about the following suggestion:
```diff
diff --git a/pkg/component/autoscaling/gardenercustommetrics/service.go b/pkg/component/autoscaling/gardenercustommetrics/service.go
index 4e8bfa955..9fb51dc1e 100644
--- a/pkg/component/autoscaling/gardenercustommetrics/service.go
+++ b/pkg/component/autoscaling/gardenercustommetrics/service.go
@@ -25,9 +25,9 @@ import (
 const serviceName = "gardener-custom-metrics"
 
 func (gcmx *gardenerCustomMetrics) service() *corev1.Service {
-	//This service intentionally does not contain a pod selector. As a result, KCM does not perform any endpoint management.
-	//Endpoint management is instead done by the gardener-custom-metrics leader instance, which ensures a single endpoint,
-	//directing all traffic to the leader.
+	// This service intentionally does not contain a pod selector. As a result, KCM does not perform any endpoint management.
+	// Endpoint management is instead done by the gardener-custom-metrics leader instance, which ensures a single endpoint,
+	// directing all traffic to the leader.
 	return &corev1.Service{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      serviceName,
```

Reply:
Thanks! Some automatic tool sneaked behind my back and reformatted this.
Done.